module/VuFind/tests/unit-tests/src/VuFindTest/ILS/Driver/AlmaTest.php

```diff
@@ -197,6 +204,108 @@ public function testGetHolding(): void
         );
     }
 
+    /**
+     * Data provider for testing getMyProfile
+     *
+     * @return Generator
+     */
+    public static function getTestGetMyProfileData(): \Generator
+    {
+        yield 'Address 2 not set' => [
+            'fixtureKey' => 'getMyProfile test 1',
+            'expected' => [
+                'firstname' => 'John',
+                'lastname' => 'Smith',
+                'birthdate' => '',
+                'address1' => 'A street 1',
+                'address2' => '',
+                'city' => 'Far away',
+                'country' => 'Far',
+                'zip' => '00000',
+                'phone' => '0123456789',
+                'mobile_phone' => null,
+                'expiration_date' => null,
+                'group' => 'prefix_test',
+                'address3' => 'Not a default field',
+                'group_code' => 'test',
+                'email' => null,
+                'home_library' => null,
+            ],
+        ];
+
+        yield 'Missing most fields' => [
+            'fixtureKey' => 'getMyProfile test 2',
+            'expected' => [
+                'firstname' => '',
+                'lastname' => 'Smith',
+                'birthdate' => '',
+                'address1' => '',
+                'address2' => '',
+                'city' => '',
+                'country' => '',
+                'zip' => '',
+                'phone' => '',
+                'mobile_phone' => null,
+                'expiration_date' => null,
+                'group' => 'prefix_test',
+                'address3' => '',
+                'group_code' => 'test',
+                'email' => null,
+                'home_library' => null,
+            ],
+        ];
+    }
+
+    /**
+     * Test getMyProfile
+     *
+     * @param string $fixtureKey Key which selects the correct test path in json file containing
+     *                           multiple tests.
+     * @param array  $expected   Expected results for the test
+     *
+     * @return       void
+     * @dataProvider getTestGetMyProfileData
+     */
+    public function testGetMyProfile(string $fixtureKey, array $expected): void
+    {
+        $adjustedConfig = $this->defaultDriverConfig;
+        $adjustedConfig['Catalog']['translationPrefix'] = 'prefix_';
+        $this->createConnector('get-my-profile', $adjustedConfig, $fixtureKey);
+        $result = $this->driver->getMyProfile(['id' => '1111']);
+        if ($result['group']) {
+            // Alma uses Translatable strings in group field so do a truth check here to assure result.
+            $this->assertTrue($result['group'] instanceof TranslatableString);
```

Comment:
I believe you can simplify to:

```suggestion
            // Alma uses Translatable strings in group field; make sure that passed through correctly.
            $this->assertInstanceOf(TranslatableString::class, $result['group']);
```